pkg/receive/handler.go

```diff
@@ -747,34 +758,65 @@ func (h *Handler) fanoutForward(pctx context.Context, tenant string, wreqs map[e
 	// This is needed if context is canceled or if we reached success of fail quorum faster.
 	defer func() {
 		go func() {
-			for err := range ec {
+			for errReplicaFn := range ec {
+				_, _, err := errReplicaFn()
 				if err != nil {
 					level.Debug(tLogger).Log("msg", "request failed, but not needed to achieve quorum", "err", err)
 				}
 			}
 		}()
 	}()
 
-	var success int
+	var (
+		// perReplicaSuccess is used to determine if enough requests succeeded for given replica number.
+		preReplicaSuccess = make([]int, h.options.ReplicationFactor)
+		// replicaMultiError contains all requests errors per replica.
+		replicaMultiError = make([]errutil.MultiError, h.options.ReplicationFactor)
```

Comment:
So the variable could be named "replicaReqErrors", right?

Reply:
Fixed :+1: